### PR TITLE
feat: reenabled import checking rules (eslint-plugin-import)

### LINKED ISSUE #15: Add `eslint-plugin-import`
Either when https://github.com/import-js/eslint-plugin-import/issues/2948 is fixed or switch to https://github.com/un-ts/eslint-plugin-import-x

### PR DESCRIPTION
**Pull request by jerome-benoit**
closes #15

`eslint-import-resolver-typescript` dep is required to make import module resolution import works in TS files.

### CODE REVIEW

File: lib/ts.js [hunk 1]
```diff
@@ -88,6 +88,12 @@ function typescriptify (configs, options) {
       // @ts-ignore - Plugin type mismatch between typescript-eslint and ESLint
       '@typescript-eslint': plugin,
     },
+    settings: {
+      'import/resolver': {
+        typescript: true,
+        node: true,
+      },
+    },
```
Comment (voxpelli) [inline lib/ts.js:96]: How does it know where to import it from when its imported like this? 🤔

Comment (voxpelli) [reply lib/ts.js:96]: Realized I asked the same back in October and we had a thread about it 🙈 https://github.com/neostandard/neostandard/pull/196#discussion_r1818146418

**Review by voxpelli — APPROVED** [review]

**Comment by tagliala** [comment]
Hello,

After this change, I do not need

https://github.com/diowa/ruby3-rails7-bootstrap-heroku/blob/c6dc3387c597e03b1916cdec2ebfbc0dc79eb7cd/package.json#L52

anymore, correct?

**Comment by voxpelli** [comment]
@tagliala That's correct! And if you still want additional rules then you should use the plugin instance that neostandard re-exports 🙂 

I will do a release soon: https://github.com/neostandard/neostandard/pull/227

**Comment by tagliala** [comment]
Thanks, I'll proceed to remove that line after the new release

I'm using some additional rules in other projects for additional rules, we have something like this

```js
import neostandard from 'neostandard'

export default [
  {
    ignores: [
      'app/assets/config/manifest.js',
      'app/assets/javascript/**/vendor/*.js',
      'config/**/*.js',
    ],
  },
  ...neostandard(),
  {
    rules: {
      'object-shorthand': 'off',
      'comma-dangle': ['error', {
          'arrays': 'always-multiline',
          'objects': 'always-multiline',
          'imports': 'never',
          'exports': 'never',
          'functions': 'never'
      }]
    },
    settings: {
      'import/resolver': {
        node: {
          paths: ['app/javascript'],
        }
      }
    }
  }
]
```

Is this the correct approach?

**Comment by voxpelli** [comment]
Have a follow up PR before releasing: https://github.com/neostandard/neostandard/pull/231

@tagliala

This would be a good config for you:

```js
import neostandard, { resolveIgnoresFromGitignore } from 'neostandard'

export default neostandard({
  ignores: [
    ...resolveIgnoresFromGitignore(), // Would be a good addition
    'app/assets/config/manifest.js',
    'app/assets/javascript/**/vendor/*.js',
    'config/**/*.js',
  ],
});
```

Your overrides are not really necessary:

* `object-shorthand`: is only a `warn`: https://github.com/neostandard/neostandard/blob/5828bac6392c00422120e0dae65c0bc3a991be07/lib/configs/base.js#L33
* `@stylistic/comma-dangle` is deactivated completely: #143

If you want to do the overrides, then:

```js
import neostandard, { resolveIgnoresFromGitignore } from 'neostandard'

export default [
  ...neostandard({
    ignores: [
      ...resolveIgnoresFromGitignore(), // Would be a good addition
      'app/assets/config/manifest.js',
      'app/assets/javascript/**/vendor/*.js',
      'config/**/*.js',
    ],
  }),
  {
    name: 'tagliala/overrides',
    rules: {
      'object-shorthand': 'off',
      // This is the correct name for that rule
      '@stylistic/comma-dangle': ['error', {
        arrays: 'always-multiline',
        objects: 'always-multiline',
        imports: 'never',
        exports: 'never',
        functions: 'never'
      }],
    },
  },
]
```

**Comment by voxpelli** [comment]
You can then use https://github.com/eslint/config-inspector to inspect that the config is working as intended

**Comment by tagliala** [comment]
thanks, great resources

**Comment by tagliala** [comment]
I'm taking a look

Maybe this needs to be fixed upstream

image

```
@stylistic/jsx-indent Enforce JSX indentation. Deprecated, use `indent` rule instead.
```

**Comment by voxpelli** [comment]
@tagliala  Feel free to create an issue for that :) Its defined here: https://github.com/neostandard/neostandard/blob/acd8da9f00f97fcb1c3e506de76abc962df4acb8/lib/configs/jsx.js#L87-L90

**Pull request merged**